### PR TITLE
updates to BGE joint calling pipeline from GVS tieout

### PR DESCRIPTION
**Pull request by meganshand**
While tying out the BGE joint calling pipeline on GVS, two differences from the WARP pipeline implementation came up that should be updated here to match what we decided to do in GVS.

1. The VETS filtering model should be trained on the variants only in the target regions, but should be applied to all variants output by the pipeline (including the hard filtered ones). This is added with the --ignore-all-filters optional argument which is shown in the bge inputs json.
2. We added the ability to add a description to the OUTSIDE_OF_TARGETS filter for the header of the output VCF in GATK 4.6.0.0 which can now be added to the `HardFilterAndMakeSitesOnly` task if the targets are provided as an optional input.

This is expected to change one scientific test (the bge specific run of JointGenotyping), but all plumbing tests should pass.

### CODE REVIEW

**Comment by meganshand** [comment]
Scientific test for JointGenotyping is here: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/84485/

The only failure is the expected BGE differences and I looked at the output and all the differences were as expected.

File: pipelines/broad/dna_seq/germline/joint_genotyping/by_chromosome/JointGenotypingByChromosomePartTwo.changelog.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 1.5.1
+2024-09-10 (Date of Last Commit)
+
+* Update to BGE filtering options in JointCalling. This has no effect on this pipeline.
```
Comment (ekiernan) [inline pipelines/broad/dna_seq/germline/joint_genotyping/by_chromosome/JointGenotypingByChromosomePartTwo.changelog.md:4]:
```suggestion
* Update to BGE filtering options in JointCalling; this has no effect on this pipeline
```

File: pipelines/broad/dna_seq/germline/joint_genotyping/by_chromosome/JointGenotypingByChromosomePartOne.changelog.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 1.5.1
+2024-09-10 (Date of Last Commit)
+
+* Update to BGE filtering options in JointCalling. This has no effect on this pipeline.
```
Comment (ekiernan) [inline pipelines/broad/dna_seq/germline/joint_genotyping/by_chromosome/JointGenotypingByChromosomePartOne.changelog.md:4]:
```suggestion
* Update to BGE filtering options in JointCalling; this has no effect on this pipeline

File: pipelines/broad/dna_seq/germline/joint_genotyping/UltimaGenomics/UltimaGenomicsJointGenotyping.changelog.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+# 1.2.1
+2024-09-10 (Date of Last Commit)
+
+* Update to BGE filtering options in JointCalling. This has no effect on this pipeline.
```
Comment (ekiernan) [inline pipelines/broad/dna_seq/germline/joint_genotyping/UltimaGenomics/UltimaGenomicsJointGenotyping.changelog.md:4]:
```suggestion
* Update to BGE filtering options in JointCalling; this has no effect on this pipeline

**Pull request merged**